### PR TITLE
build: fix esBuild only emitting esm format

### PR DESCRIPTION
**Pull request by WenheLI**
This PR fixes the #520 which only generates `esm` format for nodejs.
To make a temporary workaround, I imported `babel` to transform `esm` to `cjs`.
For the build performance, despite using `esbuild` + `babel`, the speed is still out-perform `tsc`:
| tsc | esBuild+Babel | esBuild |
 | - | - | -|
|11.0s | 2.8s | 1.8s |

Note: since `esbuild` omits `interface` in ts, we need to operate a regular `build` prior to `dev-build` to avoid missing `interface` only files.

In the long run, I will add try to add the support of `esm2cjs` transforming for `esBuild` to avoid the babel overhead.

### CODE REVIEW

File: packages/app/package.json [hunk 1]
```diff
@@ -10,8 +10,9 @@
     "test": "ts-node ../../run_tests.ts",
     "build": "npm run clean && npm run compile",
     "dev-build": "npm run clean && npm run dev-compile",
+    "dev-trans": "babel --plugins @babel/plugin-proposal-export-namespace-from ./dist/ --out-dir ./dist && babel --plugins @babel/plugin-transform-modules-commonjs ./dist/ --out-dir ./dist",
     "clean": "rm -rf ./dist && rm -rf tsconfig.tsbuildinfo",
-    "dev-compile": "esbuild ./src/*.ts ./src/**/*.ts --tsconfig=tsconfig.json --outdir=dist && sh setup.sh",
+    "dev-compile": "esbuild ./src/*.ts ./src/**/*.ts --tsconfig=tsconfig.json --outdir=dist && npm run dev-trans && sh setup.sh",
```
Comment (yorkie) [inline packages/app/package.json:15]: Shall we have a universal script for wrapping the dev-trans and dev-compile?

Comment (WenheLI) [reply packages/app/package.json:15]: For now, this `dev-trans` is just a temporary workaround. We will get rid of it eventually. In this case, I don't think we need to have a universal script.

**Comment by yorkie** [comment]
@WenheLI The current script looks too complicated, may I ask you to simplify this?

**Comment by WenheLI** [comment]
> @WenheLI The current script looks too complicated, may I ask you to simplify this?

esBuild now support directly exporting packages to `cjs`, which means we do not need to manually transform it using `babel`.
See https://github.com/evanw/esbuild/issues/109#event-3734610089, I will update the `esBuild` once it is officially launched.

**Comment by yorkie** [comment]
@WenheLI So the esbuild version update is the only change to fix #520?

**Comment by WenheLI** [comment]
> @WenheLI So the esbuild version update is the only change to fix #520?

Yep, we only need to wait for the official launch.

**Pull request closed without merging**